spark/src/main/java/org/apache/iceberg/spark/actions/ConvertEqDeletesStrategy.java

```diff
@@ -0,0 +1,176 @@
+
+public class ConvertEqDeletesStrategy implements RewriteDeleteStrategy {
+  private static final Logger LOG = LoggerFactory.getLogger(ConvertEqDeletesStrategy.class);
+
+  private final Table table;
+  private long deleteTargetSizeInBytes;
+  private int splitLookback;
+  private long splitOpenFileCost;
+
+  private CloseableIterable<FileScanTask> tasksWithEqDelete;
+  private Iterable<DeleteFile> deletesToReplace;
+  private final JavaSparkContext sparkContext;
+
+  /**
+   * Defines whether to split out the result position deletes by data file names.
+   *
+   * This should be used in EqualityDeleteRewriter.
+   */
+  public static final String SPLIT_POSITION_DELETE = "split-position-delete";
+
+  public ConvertEqDeletesStrategy(SparkSession spark, Table table) {
+    this.table = table;
+    this.sparkContext = JavaSparkContext.fromSparkContext(spark.sparkContext());
+    this.deleteTargetSizeInBytes = PropertyUtil.propertyAsLong(
+        table.properties(),
+        TableProperties.DELETE_TARGET_FILE_SIZE_BYTES,
+        TableProperties.WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT);
+    this.splitLookback = PropertyUtil.propertyAsInt(
+        table.properties(),
+        TableProperties.SPLIT_LOOKBACK,
+        TableProperties.SPLIT_LOOKBACK_DEFAULT);
+    this.splitOpenFileCost = PropertyUtil.propertyAsLong(
+        table.properties(),
+        TableProperties.SPLIT_OPEN_FILE_COST,
+        TableProperties.SPLIT_OPEN_FILE_COST_DEFAULT);
+  }
+
+  @Override
+  public String name() {
+    return "CONVERT-EQUALITY-DELETES";
+  }
+
+  @Override
+  public Table table() {
+    return table;
+  }
+
+  @Override
+  public Iterable<DeleteFile> selectDeletes() {
+    CloseableIterable<FileScanTask> fileScanTasks = null;
+    try {
+      fileScanTasks = table.newScan()
+          .ignoreResiduals()
+          .planFiles();
+    } finally {
+      try {
+        if (fileScanTasks != null) {
+          fileScanTasks.close();
+        }
+      } catch (IOException ioe) {
+        LOG.warn("Failed to close task iterable", ioe);
+      }
+    }
+
+    tasksWithEqDelete = CloseableIterable.filter(fileScanTasks, scan ->
+        scan.deletes().stream().anyMatch(delete -> delete.content().equals(FileContent.EQUALITY_DELETES))
+    );
+
+    Set<DeleteFile> eqDeletes = Sets.newHashSet();
+    tasksWithEqDelete.forEach(task -> {
+      eqDeletes.addAll(task.deletes().stream()
+          .filter(deleteFile -> deleteFile.content().equals(FileContent.EQUALITY_DELETES))
+          .collect(Collectors.toList()));
+    });
+
+    deletesToReplace = eqDeletes;
+
+    return deletesToReplace;
+  }
+
+  @Override
+  public Iterable<DeleteFile> rewriteDeletes() {
+    Map<StructLikeWrapper, Collection<FileScanTask>> groupedTasks =
+        TableScanUtil.groupTasksByPartition(table.spec(), tasksWithEqDelete.iterator());
+
+    // Split and combine tasks under each partition
+    List<Pair<StructLike, CombinedScanTask>> combinedScanTasks = groupedTasks.entrySet().stream()
```

Comment:
After reading this, I think we can make the `RewriteDeleteStrategy` interface closer to `RewriteStrategy` interface. What we have here is basically the equivalent of `planFileGroups` plus `rewriteFiles` in `RewriteStrategy`. So I would propose we have the following methods in `RewriteDeleteStrategy` to be more aligned:

```
Iterable<DeleteFile> selectDeletesToRewrite(Iterable<FileScanTask> dataFiles);

Iterable<List<FileScanTask>> planDeleteGroups(Iterable<DeleteFile> deleteFiles);

Set<DeleteFile> rewriteDeletes(List<DeleteFile> deleteFilesToRewrite);
```

And we can get the partition `StructLike` directly from the list of scan tasks instead of passing it through the task pair in `EqualityDeleteRewriter`. In this way, we can also enable partial progress for commits.

Reply:
I'm updating this PR according to the API changes, the changes of `selectDeletesToRewrite` and `rewriteDeletes` are OK to me. But `Iterable<List<FileScanTask>> planDeleteGroups(Iterable<DeleteFile> deleteFiles);` is a bit wired since it returns groups of `List<FileScanTask>`, while a `FileScanTask` could contains several deletes which don't exist in the `deleteFiles`. So I prefer to return `Iterable<list<DeleteFile>>`. It is worth noting that one data file could have several deletes,  so we could not directly using `FileScanTask` to transfer the deletes. This is slightly different from the date file rewrite.

> And we can get the partition StructLike directly from the list of scan tasks instead of passing it through the task pair in EqualityDeleteRewriter. In this way, we can also enable partial progress for commits.

The scan tasks in a group may belong to different partitions. So unless we group deletes by partition, it needs to know the partition values.